### DISCUSSION TITLE
Can we have an arm64 docker image please ?

### DISCUSSION THREAD

**Discussion by boul2gom** [outline/outline#3249] [Self Hosting]
There is a lot of servers running under arm64 architecture, so it can be nice if u can add a Dockerfile for the arm64 architecture. Thanks

**Comment by kamelohr** [comment]
I totally back this request!

And funny enough as I'm writing this post, I'm building an arm64 image myself in order to run it on a raspberrypi based k3s cluster (for personal use if you're wondering).

In case anybody is interested how to do that:

Requirements: [Have a (local) docker registry server available](https://hub.docker.com/_/registry)

1. Clone the outline git repo
2. Modify lines 9 to 17 in `Dockerfile` to
```
# ---
FROM deps-common AS deps-dev
RUN yarn install --no-optional --frozen-lockfile --network-timeout 1000000 && \
  yarn cache clean

# ---
FROM deps-common AS deps-prod
RUN yarn install --production=true --frozen-lockfile --network-timeout 1000000 && \
  yarn cache clean
```
Why? I ran into some timeouts and had to extend the commands.

3. Build the arm64 Image with:
`docker buildx build --platform linux/arm64 -t <your-registry>/outline/outline:latest -t <your-registry>/outline/outline:<latest-version-tag> --load .`

4. Push the image to your registry
`docker push <your-registry>/outline/outline:latest`
`docker push <your-registry>/outline/outline:<latest-version-tag> `

5. In the docker-compose.yml of your deployment replace the `image:` value with
`<your-registry>/outline/outline:latest` 

Did this on an 2019 Intel based Mac.

**Reply by boul2gom** [reply]
Thanks for your reply, but i have opened a PR to add multiples platforms : https://github.com/outline/outline/pull/3262 😂

**Reply by kamelohr** [reply]
Ah that's cool. Didn't see that. 
Tbh, I'm not that pro in engineering/dev regards and just wanted to share a quick solution for the time the hub image isn't providing other platforms 🙂

**Reply by boul2gom** [reply]
Sure, its a really fast working solution ^^ Btw i think you can also do that without custom registry, because  docker images can be stored locally so you can only build it, without pushing on any registry

**Reply by kamelohr** [reply]
Thanks for your hint about local images. In my case however that was on purpose because I wanted to run the image on an raspberrypi and not the (beefier) machine I was creating the image on. But this might be useful for somebody else though :)
